test/irmin-pack/test_pack_version_bump.ml

```diff
@@ -0,0 +1,200 @@
+
+open! Import
+open Common
+
+(** {2 Preamble} *)
+
+let src = Logs.Src.create "tests.version_bump" ~doc:"Test pack version bump"
+
+module Log = (val Logs.src_log src : Logs.LOG)
+
+module Util = struct
+  (** Following are generic utils *)
+
+  let exec_cmd = Common.exec_cmd
+  let tmp_dir () = Filename.temp_file "test_pack_version_bump_" ""
+
+  (** Make a directory; dir is assumed to not exist; parent is assumed to exist *)
+  let mkdir p =
+    assert (not (Sys.file_exists p));
+    Filename.quote_command "mkdir" [ p ] |> fun cmd ->
+    exec_cmd cmd |> function
+    | Ok () -> ()
+    | Error n ->
+        Fmt.failwith "Command `%s' exited with non-zero code %d\n" cmd n
+
+  (** Copy src to dst; dst is assumed to not exist *)
+  let copy_dir src dst =
+    assert (not (Sys.file_exists dst));
+    (* don't check if it is empty; perhaps we should *)
+    Filename.quote_command "cp" [ "-R"; src; dst ] |> fun cmd ->
+    exec_cmd cmd |> function
+    | Ok () -> ()
+    | Error n ->
+        Fmt.failwith
+          "Failed to set up test env; command `%s' exited with non-zero code %d\n"
+          cmd n
+
+  (* NOTE we try to avoid dependence on Fpath in the following, by
+     using standard OCaml functions *)
+
+  (** Identify the root directory, by comparing st_dev,st_ino *)
+  let is_root =
+    let open Unix in
+    let root_stat = stat "/" in
+    fun s ->
+      let stat = Unix.stat s in
+      (stat.st_dev, stat.st_ino) = (root_stat.st_dev, root_stat.st_ino)
+
+  (** Starting from ".", try to find a parent directory that has a given
+      property. *)
+  let find_parent_matching test =
+    let rec go path =
+      match test path with
+      | true -> Ok path
+      | false -> (
+          match is_root path with true -> Error () | false -> go (path ^ "/.."))
+    in
+    go "."
+
+  (** More specific utils from here *)
+
+  let v1_store_archive_dir = "test/irmin-pack/data/version_1"
+
+  (** Find the project root, that contains the v1_store_archive_dir *)
+  let project_root =
+    find_parent_matching (fun d ->
+        Sys.file_exists (d ^ "/" ^ v1_store_archive_dir))
+    |> function
+    | Ok s -> s
+    | Error () ->
+        Fmt.failwith
+          "Couldn't find project root containing path to %s, after examining \
+           current directory %s and ancestors"
+          v1_store_archive_dir (Sys.getcwd ())
+
+  (* Given the above, the following should always succeed *)
+  let _ = assert (Sys.file_exists (project_root ^ "/" ^ v1_store_archive_dir))
+
+  module Unix_ = Irmin_pack.IO.Unix
+
+  (** Get the version of the underlying file; file is assumed to exist; file is
+      assumed to be an Irmin_pack.IO.Unix file *)
+  let io_get_version ~fn : [ `V1 | `V2 ] =
+    assert (Sys.file_exists fn);
+    let t = Unix_.v ~version:None ~fresh:false ~readonly:true fn in
+    let r = Unix_.version t in
+    Unix_.close t;
+    r
+
+  let alco_check_version ~pos ~expected ~actual =
+    (* FIXME? Version is private in Irmin_pack *)
```

Comment:
Yes, you'll need to re-expose it in `irmin_pack_intf.ml` (as mentioned in my previous comment), but that's OK.

Reply:
Done. Apologies - I did see you mention Version should be exposed, but somehow I overlooked it.